tests/python_package_test/test_sklearn.py

```diff
@@ -319,6 +322,7 @@ def test_grid_search():
     assert grid.best_estimator_.best_score_['valid_0']['error'] == 0
     assert score >= 0.2
     assert score <= 1.
+    assert evals_result == grid.best_estimator_.evals_result_
```

Comment:
This assert fails at `master`:
```
>       assert evals_result == grid.best_estimator_.evals_result_
E       AssertionError: assert {'valid_0': O..., 0, 0, 0])])} == {'valid_0': O... [0, 0, 0])])}
E         Differing items:
E         {'valid_0': OrderedDict([('multi_logloss', [0.5915701709051111, 0.5814219901209461, 0.6047065388140261, 0.555159764382..., 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])])} != {'valid_0': OrderedDict([('multi_logloss', [0.11975431959688579, 0.11424123104047454, 0.11872456319439122]), ('error', [0, 0, 0])])}
```
https://github.com/microsoft/LightGBM/runs/4499752025?check_suite_focus=true#step:3:834